### DISCUSSION TITLE
Track recording often only saves part of track

### DISCUSSION THREAD

**Discussion by Per-Nothagen** [osmandapp/OsmAnd#12990] [Q&A]
Since some times last winter I have the problem that a successfully recorded GPX track is not saved in its entirety if I select save and close (I am not sure about the exact English words since I use the Swedish version). If I instead select save and continue it works. Since the problem appeared there have been an number of major changes of the track recorder but this problem has been consistent. In one version there was a save button which presented a small window (like a toaster, Android terminology) with a "change name" link. It semed like this window appeared after the file was actually written. Still, if I ignored until it disappeared by it self this the resulting GPX was OK but if I pressed the link and changed the name only part of the track was saved. I would say that in any of the failing situations it fails 2 times of 3. If I use save and continue it never failed. Before the problems started to appear I have used the track recorder for years without problems. I wish I could have given you some more hints on some circumstances around the failure, especially since I am a programmer my self and know how important that is, but I have not found any. I have neither found any pattern in what part is missing and how much, except that it is always the end of the track that is missing but it could be half of as 8h track or 15 min on a 1.5h track. I am running it on a Galaxy A520F with Android 8.0.0. There is plenty of storage space available.

**Comment by pebogufi** [comment]
Which version of osmand do you use ?

**Reply by Per-Nothagen** [reply]
4.0.8. It updates automatically and the problem has occurred in all versions it has updated to since some times last winter. Unfortunately I did not make any notes exactly when the problem appeared. I can not find any version number for the plug in but it might be updated in sync with the OsmAnd app.

**Reply by Rovanion** [reply]
I have had this hppen repeatedly now this week on OsmAnd 4.9.10. I have never had this issue before. Swedish version here too.

**Reply by x51mon** [reply]
Might be caused by the setting "Auto-split recordings after gaps" in Plugins/Trip Recording, see https://github.com/osmandapp/OsmAnd/issues/21831#issuecomment-2625222093
